### PR TITLE
Add internal coordinates representation

### PR DESCRIPTION
**Pull request by maltewae**
Adding classes for handling coordinates in different systems (chip and stage) in LabExT. This allows easier and safer handling of the coordinate transformations.

### CODE REVIEW

File: requirements.in [hunk 1]
```diff
@@ -38,6 +38,10 @@ mkdocstrings
 # auxiliary data structures
 bidict
 
+# tools for testing
+parameterized
+pytest
+
```
Comment (marcoep) [inline requirements.in:44]: pls remove

Testing requirements will get their own requirements file in #48

File: requirements.txt [hunk 2]
```diff
@@ -1,17 +1,15 @@
 #
-# This file is autogenerated by pip-compile with python 3.7
+# This file is autogenerated by pip-compile with python 3.8
```
Comment (marcoep) [inline requirements.txt:2]: Please do not change requirements file.

Testing requirements will be updated in #48

File: LabExT/Movement/Transformations.py [hunk 3]
```diff
@@ -42,6 +44,124 @@ def stage_to_chip(self, stage_coordinate):
...
+
+    @classmethod
+    def from_numpy(cls, array: np.ndarray) -> Type[Coordinate]:
+        """
+        Returns a new coordinate created from a one-dimensional numpy array.
+        """
+        if array.ndim != 1:
+            raise ValueError("The given array is not a 1D-Array.")
+
+        return cls(*array.tolist()[:3])
```
Comment (marcoep) [inline LabExT/Movement/Transformations.py:68]: would it make sense to check for longer than 3 elem arrays? Or less than 3 elems?

File: LabExT/Movement/Transformations.py [hunk 4]
```diff
@@ -42,6 +44,124 @@ def stage_to_chip(self, stage_coordinate):
...
+    A Coordinate in the coordinate system of a stage.
+    """
+
+    def __init__(self, x=0, y=0, z=0) -> None:
+        super().__init__(x, y, z)
+
+
+class ChipCoordinate(Coordinate):
+    """
+    A Coordinate in the cooridnate system of a chip.
```
Comment (marcoep) [inline LabExT/Movement/Transformations.py:158]: typo

File: LabExT/Tests/Movement/Transformations_test.py [hunk 5]
```diff
@@ -2,13 +2,190 @@
...
+            coordinate_type.from_numpy(array), x, y, z)
+
+    def _assert_coordinate_with_correct_values(
+            self, coordinate: Type[Coordinate], x, y, z):
+        self.assertEqual(coordinate.x, x)
+        self.assertEqual(coordinate.y, y)
+        self.assertEqual(coordinate.z, z)
+
+
+class StageCooridnateTest(CoordinateTest):
```
Comment (marcoep) [inline LabExT/Tests/Movement/Transformations_test.py:93]: typo

**Review by marcoep — CHANGES_REQUESTED** [review]

File: LabExT/Movement/Transformations.py (hunk 3 above)
Comment (maltewae) [reply LabExT/Movement/Transformations.py:44]: Arrays longer than 3 elements are simply truncated and shorter than 3 are padded with zeros.

Do you want to make a check that only arrays of length 3 are allowed?

Comment (marcoep) [reply LabExT/Movement/Transformations.py:44]: I dont see the padding code. But if you pad, then its ok.

**Comment by marcoep** [comment]
Will be merged as soon as #48 is done :)

**Review by marcoep — APPROVED** [review]

File: LabExT/Movement/Transformations.py (hunk 3 above)
Comment (marcoep) [reply LabExT/Movement/Transformations.py:44]: Okay, I see. thanks so much! Now just waiting on #48

**Review by marcoep — APPROVED** [review]

**Pull request merged**